docs/topics/equality.md

```diff
@@ -1,50 +1,124 @@
+
 To provide a custom equals check implementation, override the
-[`equals(other: Any?): Boolean`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-any/equals.html) function.
-Functions with the same name and other signatures, like `equals(other: Foo)`, don't affect equality checks with
+[`equals(other: Any?): Boolean`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-any/equals.html) function:
+
+```kotlin
+class Point(val x: Int, val y: Int) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is Point) return false
+
+        // Compares properties for structural equality
+        return this.x == other.x && this.y == other.y
+    }
+}
+```
+
+Functions with the same name and other signatures (like `equals(other: Foo)`) don't affect equality checks with
 the operators `==` and `!=`.
 
 Structural equality has nothing to do with comparison defined by the `Comparable<...>` interface, so only a custom 
 `equals(Any?)` implementation may affect the behavior of the operator. 
 
 ## Referential equality
 
+Referential equality verifies the memory addresses of two objects to determine if they are the same instance.
+
 Referential equality is checked by the `===` operation and its negated counterpart `!==`. `a === b` evaluates to
-true if and only if `a` and `b` point to the same object. For values represented by primitive types at runtime
+true if and only if `a` and `b` point to the same object: 
+
+```kotlin
+fun main() {
+    var a = "Hello"
+    var b = a
+    var c = "world"
+    var d = "world"
+
+    println(a === b)
+    // true
+    println(a === c)
+    // false
+    println(c === d)
+    // true
+
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3"}
+
+For values represented by primitive types at runtime
 (for example, `Int`), the `===` equality check is equivalent to the `==` check.
 
+> For more information about equality in Kotlin/JS, see the [Kotlin/JS](js-interop.md#equality) documentation.
+>
+{type="tip"}
+
 ## Floating-point numbers equality
 
-When an equality check operands are statically known to be `Float` or `Double` (nullable or not), the check follows the 
-[IEEE 754 Standard for Floating-Point Arithmetic](https://en.wikipedia.org/wiki/IEEE_754). 
+When the equality check operands are statically known to be `Float` or `Double` (nullable or not), the check follows the 
+[IEEE 754 Standard for Floating-Point Arithmetic](https://en.wikipedia.org/wiki/IEEE_754).
 
-Otherwise, structural equality is used, which disagrees with the standard so that `NaN` is equal to itself, `NaN` is considered greater than any other element, including `POSITIVE_INFINITY`, and `-0.0` is not equal to `0.0`.
+Otherwise, structural equality is implemented. This approach disagrees with the standard, considering 
+that `NaN` is equal to itself, `NaN` is considered greater than any other element (including `POSITIVE_INFINITY`), and 
```

Comment:
"NaN is equal to itself", "NaN is greater than any other", and "-0.0 is not equal to 0.0" are all conjuncts. Better to use wording making it more clear.

Reply:
Is it clearer now?